tests/api_fastapi/core_api/routes/public/test_dag_run.py

```diff
@@ -254,3 +257,75 @@ def test_delete_dag_run_not_found(self, test_client):
         assert response.status_code == 404
         body = response.json()
         assert body["detail"] == "The DagRun with dag_id: `test_dag1` and run_id: `invalid` was not found"
+
+
+class TestGetDagRunAssetTriggerEvents:
+    def test_should_respond_200(self, test_client, dag_maker, session):
+        asset1 = Asset(uri="ds1")
```

Comment:
Can we also add a test where there are no assets for an existing dag_id and dag_run_id?

Reply:
@rawwar that case would become very similar or same like the case where there is no asset found at all. It is covered here https://github.com/apache/airflow/pull/43874/files#diff-2867320a4cbebd99c9a22b52b509dc2a8920137424ed007ce5a5edfbc78fec60R323-R331

Reply:
No, It won't raise a 404. Since dag_id and dag_run_id are valid.  Should it raise a 404 if no assets are found?

Reply:
The case you are talking about translates to a non asset run on the consumer dag. Like if someone triggers a manual run and tries to get the upstreamAssetEvents for such a run. It works as tested below with a manual run:
![image](https://github.com/user-attachments/assets/aa057254-e99f-4f54-a350-3798370180dc)


If needed I can handle that in a follow up, but I don't think it is needed as it is a pretty "specific" case

Reply:
I'm unable to mark a comment as resolved. Please mark it as resolved